app/containers/Avatar/index.tsx

```diff
@@ -21,19 +25,24 @@ const AvatarContainer = ({
 	onPress,
 	getCustomEmoji,
 	isStatic,
-	rid
-}: IAvatar): React.ReactElement => {
+	rid,
+	handleEdit,
+	isUserProfile
+}: IAvatar & { isUserProfile?: boolean; handleEdit?: () => void }): React.ReactElement => {
 	const subscription = useRef<Subscription>();
 	const [avatarETag, setAvatarETag] = useState<string | undefined>('');
+	const previousAvatarEtag = useRef<string | undefined>('');
+	const { colors } = useTheme();
 
 	const isDirect = () => type === 'd';
 
 	const server = useSelector((state: IApplicationState) => state.share.server.server || state.server.server);
 	const serverVersion = useSelector((state: IApplicationState) => state.share.server.version || state.server.version);
-	const { id, token } = useSelector(
+	const { id, token, avatarETagUser } = useSelector(
 		(state: IApplicationState) => ({
 			id: getUserSelector(state).id,
-			token: getUserSelector(state).token
+			token: getUserSelector(state).token,
+			avatarETagUser: getUserSelector(state).avatarETag
```

Comment:
This feels wrong.
`Avatar` is a generic component to render user images.

`id` and `token` are necessary to make requests.
The introduction of `ETag` for the logged in user is wrong, because this way we're going to end up having `avatarETag` on the state coming from a query and `avatarETagUser` coming from Redux (and why doesn't it need to be a ref?).
And then you need to `avatarETag={isUserProfile ? avatarETagUser : avatarETag}`.

It would be a lot more elegant to separate this into a custom hook.